server/schedule/checker/rule_checker.go

```diff
@@ -181,7 +182,7 @@ func (c *RuleChecker) addRulePeer(region *core.RegionInfo, rf *placement.RuleFit
 		c.handleFilterState(region, filterByTempState)
 		return nil, errNoStoreToAdd
 	}
-	peer := &metapb.Peer{StoreId: store, Role: rf.Rule.Role.MetaPeerRole()}
+	peer := &metapb.Peer{StoreId: store, Role: rf.Rule.Role.MetaPeerRole(), IsWitness: rf.Rule.IsWitness}
```

Comment:
Does fixBetterLocation also need it? And some HTTP interfaces, like add peer.

Reply:
> Does fixBetterLocation also need it

I think need it.

> And some HTTP interfaces, like add peer.

I'll take a look.

Reply:
Fixed, thanks!